src/main/java/com/iotiq/application/domain/MaterialParameter.java

```diff
@@ -12,8 +16,16 @@
 @AllArgsConstructor
 @NoArgsConstructor
 public class MaterialParameter {
+    @DecimalMin("0.05")
+    @DecimalMax("2.00")
     private Float thickness;
-    private Integer flexibility;
+    @DecimalMin("0.0")
+    @DecimalMax("600.00")
+    private Float weightPerUnitArea;
+    @Min(0)
+    @Max(100)
     private Integer breathability;
+    @Min(0)
+    @Max(100)
     private Integer moistureWicking;
```

Comment:
Also do these annotations work with hibernate? What I mean is when you run the application does hibernate put a min or max constraint on the columns generated from the model?

Reply:
Also also, this kind of validations would be better in requests because spring already provides validation with an interceptor that is applied to all controllers. Since we're checking the validation annotations here in the service, I think you can instead add the validation annotations in the requests and let spring do the validation.

Reply:
From the documentation:

> Supported types are:
> BigDecimal
> BigInteger
> CharSequence
> byte, short, int, long, and their respective wrappers
> Note that double and float are not supported due to rounding errors (some providers might provide some approximative support).

Reply:
I couldn't BigDecimal value restrict with the min and max annotations.